main/src/ca/uwaterloo/flix/language/phase/Namer.scala

```diff
@@ -2065,4 +2065,30 @@ object Namer {
     def addTag(s: String, n: Name.QName, t: Name.Ident): UseEnv = copy(tags = tags + (s -> (n, t)))
   }
 
+  /**
+    * Merges the given `imports` into the given import environment `ienv0`.
+    */
+  private def mergeImportEnvs(imports: List[WeededAst.Import], ienv0: ImportEnv): Validation[ImportEnv, NameError] = {
+    Validation.fold(imports, ienv0) {
+      case (ienv1, WeededAst.Import.Import(name, alias, loc)) =>
+        ienv1.imports.get(alias.name) match {
+          case None => ienv1.addImport(alias.name, name).toSuccess
+          case Some(otherName) => mkDuplicateNamePair(alias.name, loc, SourceLocation.mk(otherName.sp1, otherName.sp2))
+        }
+    }
+  }
+
+  /**
+    * Companion object for the [[ImportEnv]] class.
+    */
+  private object ImportEnv {
+    val empty: ImportEnv = ImportEnv(Map.empty)
+  }
+
+  /**
+    * Represents an environment of "imported" Java classes or interfaces
+    */
+  private case class ImportEnv(imports: Map[String, Name.JavaName]) {
```

Comment:
Oh no. This is even worse! :)

My problem with this and with type aliases is that we already have "too many types". I want meaningful types.  This is just a wrapper around a meaningful type. There is no reason not to just write `Map[String, Name.JavaName]` other than its a few chars. longer than a type alias.

Flix contains 1,822 case classes (!!!!)

If we add `ImportEnv` or even just a type alias then it raises the bar for a future student/contributor/programmer to understand what is going on. If we have ` Map[String, Name.JavaName]` then they have a chance 👍

Reply:
@mlutze For rationale

Reply:
> Oh no. This is even worse! :)

I'm so glad you think so! I hated it while I was writing it, but thought that's what you meant by avoiding type aliases.

I understand your logic, however, consider what comes next. The PR that I have queued up to follow this "plumbs" `ienv` through the namer so that it can be used within the `WeededAst.Type.Ambiguous` case of `visitType`. This works, but it involves adding an `ienv` parameter to just about every single method within the namer. This requires literally hundreds of small changes (every method needs the additional parameter, and every method call).

Right now, a typical example looks like:

```
```

Reply:
Do we have something in the contributing docs about this?

Reply:
I see we already have UseEnv, so your proposal would be consistent.

What does `UseEnv` look like?

Reply:
`UseEnv` is what I copied to create `ImportEnv` as it currently exists:

https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/Namer.scala#L2041-L2066

Reply:
Then let's keep it as is. Will merge when at a real computer.